actors/common/src/main/java/org/sunbird/learner/actors/data/security/manager/DataSecurityBackgroundActor.java

```diff
@@ -1,65 +1,93 @@
+      case "backgroundEncryption":
+        backgroundEncrypt(request);
+        break;
+      case "backgroundDecryption":
+        backgroundDecrypt(request);
+        break;
+      default:
+        onReceiveUnsupportedOperation("EsSyncBackgroundActor");
+        break;
     }
   }
 
-  private void decryptUserData(Request actorMessage) {
-    ProjectLogger.log(
-        "DecryptUserData API called by " + actorMessage.getRequest().get(JsonKey.REQUESTED_BY));
-    long start = System.currentTimeMillis();
-    Response resp = new Response();
-    resp.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
-    sender().tell(resp, self());
-    Response response =
-        cassandraOperation.getAllRecords(usrDbInfo.getKeySpace(), usrDbInfo.getTableName());
-    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+  private void backgroundEncrypt(Request request) {
+    List<Map<String, Object>> userDetails = getUserDetails(request);
     int i = 0;
-    for (Map<String, Object> userMap : userList) {
-      if (!ProjectUtil.isEmailvalid((String) userMap.get(JsonKey.EMAIL))) {
-        decryptUserDataAndUpdateDb(userMap);
+    for (Map<String, Object> userMap : userDetails) {
+      if (ProjectUtil.isEmailvalid((String) userMap.get(JsonKey.EMAIL))
+          || ((String) userMap.get(JsonKey.PHONE)).length() < 14) {
+        encryptUserDataAndUpdateDb(userMap);
         i++;
       }
     }
-    ProjectLogger.log("Total No. of user data to decrypt ::: " + i);
-    long end = System.currentTimeMillis();
-    ProjectLogger.log("total time taken by application to decrypt user data:::: " + (end - start));
+    ProjectLogger.log(
+        "EsSyncBackgroundActor:backgroundEncrypt: total number of user details encrypted is: " + i,
+        LoggerEnum.INFO);
+    syncToES(request);
+  }
+
+  private void backgroundDecrypt(Request request) {
+    List<Map<String, Object>> userDetails = getUserDetails(request);
+    int i = 0;
+    if (userDetails != null) {
+      for (Map<String, Object> userMap : userDetails) {
+        if (!ProjectUtil.isEmailvalid((String) userMap.get(JsonKey.EMAIL))
+            || !(((String) userMap.get(JsonKey.PHONE)).length() < 14)) {
+          decryptUserDataAndUpdateDb(userMap);
+          i++;
+        }
+      }
+    }
+    ProjectLogger.log(
+        "EsSyncBackgroundActor:backgroundDecrypt: total number of user details encrypted is: " + i,
+        LoggerEnum.INFO);
+    syncToES(request);
+  }
+
+  @SuppressWarnings("unchecked")
+  private List<Map<String, Object>> getUserDetails(Request request) {
+    List<String> userIds = (List<String>) request.getRequest().get(JsonKey.USER_IDs);
+    Response response =
+        cassandraOperation.getRecordsByIdsWithSpecifiedColumns(
+            JsonKey.SUNBIRD, JsonKey.USER, null, userIds);
+    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+    if (CollectionUtils.isEmpty(userList)) return null;
```

Comment:
unncessary - let the empty list pass through rather. in case of list is return type one should always prefer to return empty list rather than null